src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -300,7 +301,9 @@ def relative_to_exec_root(path):
     if self.get_options().use_barebones_logger:
       zinc_args.append('--use-barebones-logger')
 
-    compiler_bridge_classpath_entry = self._zinc.compile_compiler_bridge(self.context)
+    lock = Lock()
```

Comment:
The zinc subsystem already has a lock in it, so it might be good to synchronize inside the class instead. To do that with a memoized_method, you'd memoize an inner/private method, and call it under the lock.